core/filterconstraint.cpp

```diff
@@ -0,0 +1,939 @@
+	}
+}
+
+static double int_to_float(int i, enum filter_constraint_type type)
+{
+	const type_description *desc = get_type_description(type);
+	if (!desc)
+		return 0.0;
+	switch (desc->units) {
+	case FILTER_CONSTRAINT_NO_UNIT:
+	default:
+		return (double)i;
+	case FILTER_CONSTRAINT_LENGTH_UNIT:
+		return prefs.units.length == units::METERS ? (double)i / 1000.0 : mm_to_feet(i);
+	case FILTER_CONSTRAINT_TEMPERATURE_UNIT:
+		return prefs.units.temperature == units::CELSIUS ? mkelvin_to_C(i) : mkelvin_to_F(i);
+	case FILTER_CONSTRAINT_WEIGHT_UNIT:
+		return prefs.units.weight == units::KG ? (double)i / 1000.0 : grams_to_lbs(i);
+	case FILTER_CONSTRAINT_VOLUMETRIC_FLOW_UNIT:
+		return prefs.units.volume == units::LITER ? (double)i / 1000.0 : ml_to_cuft(i);
+	}
+}
+
+QStringList filter_constraint_type_list_translated()
+{
+	QStringList res;
+	for (size_t i = 0; i < type_descriptions_count(); ++i)
+		res.push_back(gettextFromC::tr(type_descriptions[i].translated));
+	return res;
+}
+
+QStringList filter_constraint_string_mode_list_translated()
+{
+	QStringList res;
+	for (size_t i = 0; i < string_mode_descriptions_count(); ++i)
+		res.push_back(gettextFromC::tr(string_mode_descriptions[i].translated));
+	return res;
+}
+
+QStringList filter_constraint_range_mode_list_translated()
+{
+	QStringList res;
+	for (size_t i = 0; i < range_mode_descriptions_count(); ++i)
+		res.push_back(gettextFromC::tr(range_mode_descriptions[i].translated));
+	return res;
+}
+
+QStringList filter_constraint_range_mode_list_translated_date()
+{
+	QStringList res;
+	for (size_t i = 0; i < range_mode_descriptions_count(); ++i)
+		res.push_back(gettextFromC::tr(range_mode_descriptions[i].translated_date));
+	return res;
+}
+
+QStringList filter_constraint_negate_list_translated()
+{
+	return { gettextFromC::tr(negate_description[false]),
+		 gettextFromC::tr(negate_description[true]) };
+}
+
+// Currently, we support only two multiple choice items. Hardcode them.
+static bool filter_constraint_is_multiple_choice(enum filter_constraint_type type)
+{
+	return type == FILTER_CONSTRAINT_DIVE_MODE || type == FILTER_CONSTRAINT_DAY_OF_WEEK;
+}
+
+QStringList filter_contraint_multiple_choice_translated(enum filter_constraint_type type)
+{
+	// Currently, we support only two multiple choice lists. Hardcode them.
+	if (type == FILTER_CONSTRAINT_DIVE_MODE) {
+		QStringList types;
+		for (int i = 0; i < NUM_DIVEMODE; i++)
+			types.append(gettextFromC::tr(divemode_text_ui[i]));
+		return types;
+	} else if (type == FILTER_CONSTRAINT_DAY_OF_WEEK) {
+		// I can't wrap my head around the fact that Sunday is the
+		// first day of the week, but that's how it is.
+		return QStringList {
+			gettextFromC::tr("Sunday"),
+			gettextFromC::tr("Monday"),
+			gettextFromC::tr("Tuesday"),
+			gettextFromC::tr("Wednesday"),
+			gettextFromC::tr("Thursday"),
+			gettextFromC::tr("Friday"),
+			gettextFromC::tr("Saturday"),
```

Comment:
I thought we already had that somewhere... but it appears to be only the 'short' strings (in `save-git.c`)
That said, do we want the long strings?

Reply:
Personally, I find the long names more friendly, but in the end don't care.